relay/metadata_manager.go

```diff
@@ -0,0 +1,169 @@
+package relay
+
+import (
+	"context"
+	"fmt"
+	"github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/disperser/common/v2/blobstore"
+	"github.com/Layr-Labs/eigenda/relay/cache"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"sync/atomic"
+)
+
+// Metadata about a blob. The relay only needs a small subset of a blob's metadata.
+// This struct adds caching and threading on top of blobstore.BlobMetadataStore.
+type blobMetadata struct {
+	// the size of the blob in bytes
+	blobSizeBytes uint32
+	// the size of the file containing the encoded chunks
+	totalChunkSizeBytes uint32
+	// the fragment size used for uploading the encoded chunks
+	fragmentSizeBytes uint32
+}
+
+// metadataManager encapsulates logic for fetching metadata for blobs. Utilized by the relay Server.
+type metadataManager struct {
```

Comment:
Would it be more accurate to name these XXManagers as XXReader or XXFetcher/XXAccessor? It appears no management work needed

Reply:
Starting an offline conversation with you on naming, will update naming once we arrive at a mutually acceptable naming schema.